### ISSUE TITLE
[2.0.0] Local ansible does not work with bastion - ssh-keyscan issue

### ISSUE THREAD

**Issue by taylorsmithgg** [radekg/terraform-provisioner-ansible#50]
The local provisioner will infinitely loop `ssh-keyscan` when going through a bastion due to disregard for local ssh config.

**Comment by radekg** [comment]
Would you mind sharing the relevant `plays` and `connection` sections, minus the confidential information?

**Comment by taylorsmithgg** [comment]
``` 
provisioner "ansible" {
    connection {
      type = "ssh"
      host = "${self.private_ip}"
      user = "ec2-user"
      bastion_host = "${var.bastion}"
      bastion_user = "${var.user}"
    }

    plays {
      playbook = "${var.ansible_path}/playbooks/application/server.yaml"
    }

    local  = "yes"
    verbose = "yes"
  }
```

**Comment by taylorsmithgg** [comment]
`ssh-keyscan` does not respect `~/.ssh/config`, therefore, if you are using a bastion host to proxy to private ip addresses, it will run until timeout

**Comment by radekg** [comment]
That’s what I was able to establish. Could you possibly share at lesst an overview of what do you have in your ~/.ssh/config? Maybe it would be possible to bend https://github.com/szuecs/go-ssh-keyscan/blob/master/go-ssh-keyscan.go to make this work?

**Comment by taylorsmithgg** [comment]
```
Host aws-bastion-sandbox aws-bastion-sandbox-batch
  HostName 0.0.0.0
  IdentityFile ~/.ssh/private_key
  User user
  StrictHostKeyChecking no
  ForwardAgent yes

Host 10.139.* *.us-west-2.*
  ProxyJump aws-bastion-sandbox
  ServerAliveInterval 60
  TCPKeepAlive yes
  ControlMaster auto
  ControlPath ~/.ssh/mux-%r@%h:%p
  ControlPersist 8m
  IdentityFile ~/.ssh/private_key
  User ec2-user
  StrictHostKeyChecking no
  ForwardAgent yes
```

**Comment by taylorsmithgg** [comment]
The issue is tied up in the fact that `ssh-keyscan` doesn't consider the ssh configuration for the local system.

**Comment by mcanevet** [comment]
Any news regarding this issue?

**Comment by radekg** [comment]
Hey @taylorsmithgg and @mcanevet. Would it be possible to receive an MVP fully outlining the problem? I am not sure if I can reliably identify the problem here.

**Comment by radekg** [comment]
This will be resolved in an upcoming 2.0.0 release. No work done yet but a solution is known. PR will be linked.

**Comment by radekg** [comment]
Please refer to https://github.com/radekg/terraform-provisioner-ansible/pull/84 for details how this can be handled.
